packages/hydrogen/src/csp/csp.ts

```diff
@@ -148,12 +148,18 @@ function createCSPHeader(
     combinedDirectives.scriptSrc instanceof Array &&
     !combinedDirectives.scriptSrc.includes(nonceString)
   ) {
-    combinedDirectives.scriptSrc.push(nonceString);
+    combinedDirectives.scriptSrc = [
+      ...combinedDirectives.scriptSrc.filter((ss) => !ss.startsWith('nonce')),
+      nonceString,
+    ];
   } else if (
     combinedDirectives.defaultSrc instanceof Array &&
     !combinedDirectives.defaultSrc.includes(nonceString)
   ) {
-    combinedDirectives.defaultSrc.push(nonceString);
+    combinedDirectives.defaultSrc = [
+      ...combinedDirectives.defaultSrc.filter((ss) => !ss.startsWith('nonce')),
+      nonceString,
```

Comment:
Do we still need `!combinedDirectives.defaultSrc.includes(nonceString)` in the `if clause?

These 2 lines essentially removes all nonces and adds back the current nonce